bundle/deploy/check_running_resources.go

```diff
@@ -0,0 +1,142 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/databricks/databricks-sdk-go/service/pipelines"
+	"github.com/hashicorp/terraform-exec/tfexec"
+	tfjson "github.com/hashicorp/terraform-json"
+	"golang.org/x/sync/errgroup"
+)
+
+type ErrResourceIsRunning struct{}
+
+func (ErrResourceIsRunning) Error() string {
+	return "resource is running"
+}
+
+type checkRunningResources struct {
+}
+
+func (l *checkRunningResources) Name() string {
+	return "check-running-resources"
+}
+
+func (l *checkRunningResources) Apply(ctx context.Context, b *bundle.Bundle) error {
+	if !b.Config.Bundle.FailIfRunning {
+		return nil
+	}
+
+	tf := b.Terraform
+	if tf == nil {
+		return fmt.Errorf("terraform not initialized")
+	}
+
+	err := tf.Init(ctx, tfexec.Upgrade(true))
+	if err != nil {
+		return fmt.Errorf("terraform init: %w", err)
+	}
+
+	state, err := b.Terraform.Show(ctx)
+	if err != nil {
+		return err
+	}
+
+	isRunning := isAnyResourceRunning(ctx, b.WorkspaceClient(), state)
+	if isRunning {
+		return errors.New("some of the bundle resources are still running, deployment aborted")
+	}
+
+	return nil
+}
+
+func CheckRunningResource() *checkRunningResources {
+	return &checkRunningResources{}
+}
+
+func isAnyResourceRunning(ctx context.Context, w *databricks.WorkspaceClient, state *tfjson.State) bool {
+	if state.Values == nil || state.Values.RootModule == nil {
+		return false
+	}
+
+	errs, errCtx := errgroup.WithContext(ctx)
+
+	for _, resource := range state.Values.RootModule.Resources {
+		// Limit to resources.
+		if resource.Mode != tfjson.ManagedResourceMode {
+			continue
+		}
+
+		value, ok := resource.AttributeValues["id"]
+		if !ok {
+			continue
+		}
+		id, ok := value.(string)
+		if !ok {
+			continue
+		}
+
+		switch resource.Type {
+		case "databricks_job":
+			errs.Go(func() error {
+				isRunning, err := IsJobRunning(errCtx, w, id)
+				// If there's an error retrieving the job, we assume it's not running
+				if err != nil {
+					return err
+				}
+				if isRunning {
+					return &ErrResourceIsRunning{}
```

Comment:
Can we annotate the error with which resource is running? It'll help customers understand exactly why their deployment failed. Since we are checking all pipelines and jobs, we can collect all the URLs and print that in the error.